src/main/java/com/example/bookstore/repository/CartItemRepository.java

```diff
@@ -1,7 +1,14 @@
 package com.example.bookstore.repository;
 
 import com.example.bookstore.model.CartItem;
+import java.util.Optional;
+import java.util.Set;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 public interface CartItemRepository extends JpaRepository<CartItem, Long> {
+    Optional<CartItem> findCartItemById(Long id);
+
+    @Query(value = "SELECT * FROM cart_item c WHERE c.shopping_cart_id = :id", nativeQuery = true)
```

Comment:
is this native query required? You even can remove `@Query` as Spring would parse query from the method name